src/util/alloc/object_ref_guard.rs

```diff
@@ -20,46 +20,72 @@ use crate::util::Address;
 use crate::vm::ObjectModel;
 use crate::vm::VMBinding;
 
-/// Adjust limit for thread local buffer to make sure that we will not allocate objects whose object reference may
+/// Adjust start and end for thread local buffer to make sure that we will not allocate objects whose object reference may
```

Comment:
> we will not allocate objects whose object reference may be in another chunk.

It is totally ok to have objects like this. JikesRVM always does that. If you'd like to get the chunk owning the object, you should use `VMObjectModel::ref_to_address`. `object.to_address()` is incorrect for this case.

Reply:
This is removed in https://github.com/mmtk/mmtk-core/pull/699.